.github/workflows/doc-build.yml

```diff
@@ -1,95 +1,69 @@
----
-# Workflow to build the documentation.
-name: Build documentation
+name: Build and Deploy Sphinx Docs
+
 on:
-    push:
-        branches:
-        -   main
-    pull_request:
-        branches:
-        -   '*'
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+permissions:
+  contents: write
+  pages: write
```

Comment:
I don't think this is necessary, as the currently running deployment doesn't have it. And as you said, I'd prefer to give minimal permissions if possible !
```suggestion
```